React/game/src/app/screen/Game/index.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
+
+import Board from './Board';
+import styles from './styles.scss';
+import { serialWin } from './constants';
+
+class Game extends Component {
+  state = {
+    moves: 0,
+    gameEnd: false,
+    turn: 'X',
+    turnWinner: '',
+    history: [
+      {
+        board: Array(9).fill(''),
+        id: 0
+      }
+    ]
+  };
+
+  playerOne = 'X';
+  playerTwo = 'O';
+
+  mouseClick = place => {
+    const { history, moves } = this.state;
+    const newhistory = history.slice(0, moves + 1);
+    const current = newhistory[newhistory.length - 1];
+    const board = current.board.slice();
+
+    if (board[place] === '' && !this.state.gameEnd) {
+      board[place] = this.state.turn;
+      this.setState({
+        history: history.concat([
+          {
+            board,
+            id: history.length
+          }
+        ]),
+        turn: this.state.turn === this.playerOne ? this.playerTwo : this.playerOne,
+        moves: ++this.state.moves
+      });
+    }
+    const result = this.winner();
+    if (result === 'X') {
+      this.setState({
+        gameEnd: true,
+        turnWinner: result
+      });
+    } else if (result === 'O') {
+      this.setState({
+        gameEnd: true,
+        turnWinner: result
+      });
+    }
+  };
+
+  winner = () => {
+    const { history } = this.state;
+    const { moves } = this.state;
+    const board = history[moves - 1].board;
+    for (let i = 0; i < serialWin.length - 1; i++) {
+      if (
+        board[serialWin[i][0]] === board[serialWin[i][1]] &&
+        board[serialWin[i][1]] === board[serialWin[i][2]] &&
+        (board[serialWin[i][0]] === 'O' || board[serialWin[i][0]] === 'X')
```

Comment:
use `this.playerOne` and `this.playerTwo` here

Reply:
👍